evalml/tests/dependency_update_check/latest_dependency_versions.txt

```diff
@@ -4,35 +4,35 @@ category-encoders==2.5.1.post0
 click==8.1.7
 cloudpickle==3.0.0
 colorama==0.4.6
-dask==2023.5.0
-distributed==2023.5.0
-featuretools==1.28.0
+dask==2024.5.0
+dask-expr==1.1.0
+distributed==2024.5.0
+featuretools==1.31.0
 graphviz==0.20.3
 holidays==0.20
 imbalanced-learn==0.12.2
 ipywidgets==8.1.2
 kaleido==0.1.0
 lightgbm==4.3.0
 lime==0.2.0.1
-matplotlib==3.7.5
+matplotlib==3.9.0
 matplotlib-inline==0.1.7
-networkx==3.1
-nlp-primitives==2.11.0
```

Comment:
Why did nlp-primitives disappear?

Reply:
If I remember correctly, `latest_dependency_checker` was not happy having that in there and would fail. I'll try adding it back again and seeing what happens.

Reply:
yeah it failed again when I re-added `nlp-primitives` back into latest-dependency

Reply:
@MichaelFu512 I think you might want to try to get to the bottom of this. I think the `NaturalLanguageFeaturizer` depends on nlp-primitives for the `PolarityScore` and `DiversityScore` primitives, so it seems like `nlp-primitives` should be included here.

Reply:
I'm not sure up to date on how the dependency checkers are working in EvalML, but the latest version of nlp-primitives is `2.13.0` which was released yesterday and will be installed in your CI runs. Maybe you need to update the version from `2.11.0` to `2.13.0` here instead of removing the line?

Reply:
Same error weirdly enough,
```
Displaying dependencies which have changed, with main on the left and the new branch on the right:
21d20
< nlp-primitives==2.13.0
```

Normally, if the dependency was out of date, it'd look something like:
```
Displaying dependencies which have changed, with main on the left and the new branch on the right:
18c18
< matplotlib==3.8.4
---
> matplotlib==3.9.0
```

I'll have to check why.

Reply:
So I think the reason it failed was because of this line in `make_deps_diff.sh`

```
pip freeze | grep -v "evalml.git" | grep -E "${allow_list}" > "${DEPENDENCY_FILE_PATH}"
```

This line compares the requirements gotten from `pip freeze | grep -v "evalml.git"` to the ones in `allow_list` (which was made earlier in the file).

`allow_list` lists `nlp-primitives` while `pip freeze | grep -v "evalml.git"`  lists `nlp_primitives`. This difference makes it so that nothing even shows up in the diff.

I changed the line to:
```
(pip freeze | grep -v "evalml.git") | tr "_" "-" | grep -E "${allow_list}" > "${DEPENDENCY_FILE_PATH}"
```
which just converts the "_" to "-".